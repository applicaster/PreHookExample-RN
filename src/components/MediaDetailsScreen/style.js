import {
  Dimensions,
  StyleSheet
} from 'react-native';

const closeButtonSize = 25;
const screenWidth = Dimensions.get('window').width;

export const styles = StyleSheet.create({
  mediaDetailsScreen: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#323232',
  },
  closeButton: {
    tintColor: '#FFFFFF',
    height: closeButtonSize,
    width: closeButtonSize,
    position: 'absolute',
    top: 30,
    left: 10,
    zIndex: 5,
  },
  buttonSlop: {
    right: 5,
    bottom: 5,
  },
  imageSize: {
    width: screenWidth,
    height: screenWidth,
  },
});
