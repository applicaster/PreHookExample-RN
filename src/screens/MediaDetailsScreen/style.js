import {
  Dimensions,
  StyleSheet,
} from 'react-native';

const screenWidth = Dimensions.get('window').width;

export const styles = StyleSheet.create({
  mediaDetailsScreen: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#323232',
  },
  buttonSlop: {
    right: 5,
    bottom: 5,
  },
  imageSize: {
    width: screenWidth,
    height: screenWidth,
  },
  closeButton: {
    position: 'absolute',
    top: 45,
    right: 15,
    zIndex: 5,
  },
});
