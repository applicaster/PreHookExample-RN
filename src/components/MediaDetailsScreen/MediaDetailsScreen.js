import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  Image,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { styles } from './style';
import TransformableImage from 'react-native-transformable-image';

const CLOSE_BUTTON = 'close_button';
class MediaDetailsScreen extends Component {
  getImageSize() {
    const { imageHeight, imageWidth } = this.props;
    const screenHeight = Dimensions.get('window').height;
    const screenWidth = Dimensions.get('window').width;
    const aspectRatio = (imageWidth / imageHeight);
    let imageSize = styles.imageSize;

    if (imageWidth > imageHeight) {
      imageSize = {
        width: screenWidth,
        height: (screenWidth / aspectRatio),
      };
    } else if (imageHeight > imageWidth) {
      imageSize = {
        width: (screenHeight / aspectRatio),
        height: screenHeight, // TODO: modify after adding action buttons
      };
    }

    return imageSize;
  }

  render() {
    const { imageUrl, toggleModal } = this.props;
    const { backgroundColor, textColor = '#FFFFFF' } = this.context;
    const screenBackgroundColor = { backgroundColor, flex: 1 };
    const closeButtonColor = { tintColor: `${textColor}99` };
    return (
      <View style={[styles.mediaDetailsScreen, screenBackgroundColor]}>
        <TouchableWithoutFeedback onPress={toggleModal}>
          <Image style={[styles.closeButton, closeButtonColor]} source={{ uri: CLOSE_BUTTON }} />
        </TouchableWithoutFeedback>
        <TransformableImage style={this.getImageSize()} source={{ uri: imageUrl }} />
      </View>
    );
  }
}

MediaDetailsScreen.propTypes = {
  imageHeight: PropTypes.number,
  imageWidth: PropTypes.number,
  imageUrl: PropTypes.string,
  toggleModal: PropTypes.func,
};

MediaDetailsScreen.contextTypes = {
  backgroundColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default MediaDetailsScreen;
