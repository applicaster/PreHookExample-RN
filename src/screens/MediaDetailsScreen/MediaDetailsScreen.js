import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  View,
} from 'react-native';
import TransformableImage from '@applicaster/react-native-transformable-image';
import { sendAnalyticEvent } from 'react-native-zapp-bridge';
import { CLOSE_IMAGE_DETAIL_SCREEN } from '../../constants/analyticEvents';
import { styles } from './style';
import CloseButton from '../../buttons/CloseButton';

class MediaDetailsScreen extends Component {
  constructor(props) {
    super(props);
    this.closeModal = this.closeModal.bind(this);
  }

  getImageSize() {
    const { activeEvent } = this.props;
    const { height: imageHeight, width: imageWidth } = activeEvent.images.default;
    const screenHeight = Dimensions.get('window').height;
    const screenWidth = Dimensions.get('window').width;
    const aspectRatio = (imageWidth / imageHeight);
    const calculatedHeight = (screenWidth / aspectRatio);
    let imageSize = styles.imageSize;

    if (imageWidth > imageHeight || calculatedHeight <= screenHeight) {
      imageSize = { width: screenWidth, height: calculatedHeight };
    } else if (imageHeight > imageWidth) {
      const calculatedWidth = (screenHeight * aspectRatio);
      imageSize = { width: calculatedWidth, height: screenHeight };
    }
    
    return imageSize;
  }

  closeModal() {
    const { setNoActiveEvent, toggleModal } = this.props;
    toggleModal({ modal: 'MediaModal' });
    setNoActiveEvent();
    sendAnalyticEvent(CLOSE_IMAGE_DETAIL_SCREEN, {}).then().catch();
  }

  render() {
    const { activeEvent } = this.props;
    const { url: imageUrl } = activeEvent.images.default;
    const { backgroundColor } = this.context;
    const screenBackgroundColor = { backgroundColor, flex: 1 };
    
    return (
      <View style={[styles.mediaDetailsScreen, screenBackgroundColor]}>
        <CloseButton onPress={this.closeModal} style={styles.closeButton} />
        <TransformableImage style={this.getImageSize()} source={{ uri: imageUrl }} />
      </View>
    );
  }
}

MediaDetailsScreen.propTypes = {
  activeEvent: PropTypes.object.isRequired,
  setNoActiveEvent: PropTypes.func.isRequired,
  toggleModal: PropTypes.func.isRequired,
};

MediaDetailsScreen.contextTypes = {
  backgroundColor: PropTypes.string,
  textColor: PropTypes.string,
};

MediaDetailsScreen.defaultProps = {
  activeEvent: {
    images: {
      default: {},
    },
  },
};

export default MediaDetailsScreen;
