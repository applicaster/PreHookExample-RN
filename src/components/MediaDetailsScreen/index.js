import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  toggleModal,
} from '../../actions';
import {
  getActiveEventImageUrl,
} from '../../selectors';

import MediaDetailsScreen from './MediaDetailsScreen';

const mapStateToProps = state => ({
  imageUrl: getActiveEventImageUrl(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(MediaDetailsScreen);
