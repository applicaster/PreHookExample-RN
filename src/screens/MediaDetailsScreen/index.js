import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  setNoActiveEvent,
  toggleModal,
} from '../../actions';
import {
  getActiveEvent,
} from '../../selectors';

import MediaDetailsScreen from './MediaDetailsScreen';

const mapStateToProps = state => ({
  activeEvent: getActiveEvent(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  setNoActiveEvent,
  toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(MediaDetailsScreen);
