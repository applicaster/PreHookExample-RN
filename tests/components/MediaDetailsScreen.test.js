import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MediaDetailsScreen from '../../src/components/MediaDetailsScreen/MediaDetailsScreen';

const event = ({ width = 100, height = 100, url = 'someUrl' }) => ({
  id: 'someId',
  images: {
    default: {
      width,
      height,
      url,
    },
  },
});

describe('MediaDetailsScreen', () => {
  test('component renders correctly', () => {
    const tree = renderer.create(
      <MediaDetailsScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('image sizes', () => {
    test('component renders correctly with squared image', () => {
      const tree = renderer.create(
        <MediaDetailsScreen activeEvent={event({ height: 100, width: 100, url: 'squared' })} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('component renders correctly with landscape image', () => {
      const tree = renderer.create(
        <MediaDetailsScreen activeEvent={event({ height: 100, width: 200, url: 'lanscape' })} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('component renders correctly with vertical image with edge to edge width', () => {
      const tree = renderer.create(
        <MediaDetailsScreen activeEvent={event({ height: 200, width: 150, url: 'vertical' })} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('component renders correctly with vertical image that goes over the', () => {
      const tree = renderer.create(
        <MediaDetailsScreen activeEvent={event({ imageHeight: 5000, imageWidth: 200, imageUrl: 'supertall' })} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
