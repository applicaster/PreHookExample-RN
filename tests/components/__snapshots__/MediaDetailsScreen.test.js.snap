// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaDetailsScreen component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#323232",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      },
    ]
  }
>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "uri": "close_button",
      }
    }
    style={
      Array [
        Object {
          "height": 25,
          "left": 10,
          "position": "absolute",
          "tintColor": "#FFFFFF",
          "top": 30,
          "width": 25,
          "zIndex": 5,
        },
        Object {
          "tintColor": "#FFFFFF99",
        },
      ]
    }
    testID={undefined}
  />
  <View
    contentAspectRatio={undefined}
    enableResistance={true}
    enableScale={true}
    enableTransform={false}
    enableTranslate={true}
    maxOverScrollDistance={20}
    maxScale={1}
    onLayout={[Function]}
    onTransformGestureReleased={undefined}
    onViewTransformed={undefined}
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Image
        capInsets={
          Object {
            "bottom": 0.1,
            "left": 0.1,
            "right": 0.1,
            "top": 0.1,
          }
        }
        enableScale={true}
        enableTransform={true}
        enableTranslate={true}
        onLoad={[Function]}
        onLoadStart={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "height": 750,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
